src/dsp_tools/commands/ingest_xmlupload/bulk_ingest_client.py

```diff
@@ -82,3 +84,36 @@ def upload_file(
         print(f"{datetime.now()}: {msg}")
         logger.info(msg)
         return None
+
+    def kick_off_ingest(self) -> None:
+        """Start the ingest process on the server."""
+        url = f"{self.dsp_ingest_url}/projects/{self.shortcode}/bulk-ingest"
+        res = self.session.post(url, timeout=5)
+        if res.status_code == STATUS_CONFLICT:
+            msg = f"Ingest process on the server {self.dsp_ingest_url} is already running. Wait until it completes..."
+            print(msg)
+            logger.info(msg)
+        if res.json().get("id") != self.shortcode:
+            raise UserError("Failed to kick off the ingest process.")
+        print(f"Kicked off the ingest process on the server {self.dsp_ingest_url}. Wait until it completes...")
+        logger.info(f"Kicked off the ingest process on the server {self.dsp_ingest_url}. Wait until it completes...")
+
+    def retrieve_mapping(self) -> str | None:
+        """Try to retrieve the mapping CSV from the server."""
+        url = f"{self.dsp_ingest_url}/projects/{self.shortcode}/bulk-ingest/mapping.csv"
+        res = self.session.get(url, timeout=5)
+        if res.status_code == STATUS_CONFLICT:
+            print("Ingest process is still running. Wait until it completes...")
+            logger.info("Ingest process is still running. Wait until it completes...")
+            return None
+        elif not res.ok or not res.text.startswith("original,derivative"):
+            msg = (
+                "Dubious error while polling for the mapping CSV. "
+                f"If this happens again at the next polling, please check the logs at {logger_savepath}."
```

Comment:
What do you mean with "polling"

Reply:
https://en.wikipedia.org/wiki/Polling_(computer_science).
But perhaps I should choose another word. It's probably not well understood.